### ISSUE TITLE
An unused output of `format!()` and some other macros should produce a unused_must_use warning.

### ISSUE THREAD

**Issue by theemathas** [rust-lang/rust#126475]
### Code

```Rust
fn main() {
    format!("{}", 1);
}
```

### Current output

```Shell
No errors or warnings
```

### Desired output

```Shell
warning: unused return value of `format!()` that must be used
```

### Rationale and extra context

The `format!()` macro could probably expand to code containing [`std::hint::must_use`](https://doc.rust-lang.org/stable/std/hint/fn.must_use.html).

List of macros with similar issues:
* `cfg!()`
* `column!()`, `line!()`, `file!()`, `module_path!()`
* `concat!()`, `stringify!()`
* `env!()`, `option_env!()`
* `format!()`, `format_args!()`
* `include_bytes!()`, `include_str!()`
* `matches!()`
* `vec!()`

### Rust Version

```Shell
Nightly rust on playground: 1.81.0-nightly (2024-06-13 f1586001ace26df7cafe)
```
